library/azure_rm_aks.py

```diff
@@ -0,0 +1,560 @@
+
+                    # Cannot Update the SSH Key for now // Let service to handle it
+                    if is_property_changed('linux_profile', 'ssh_key'):
+                        self.log(("Linux Profile Diff SSH, Was {0} / Now {1}"
+                                 .format(response['linux_profile'].ssh.public_keys[0].key_data, self.linux_profile.get('ssh_key'))))
+                        to_be_updated = True
+                        # self.module.warn("linux_profile.ssh_key cannot be updated")
+
+                    # self.log("linux_profile response : {0}".format(response['linux_profile'].get('admin_username')))
+                    # self.log("linux_profile self : {0}".format(self.linux_profile[0].get('admin_username')))
+                    # Cannot Update the Username for now // Let service to handle it
+                    if is_property_changed('linux_profile', 'admin_username'):
+                        self.log(("Linux Profile Diff User, Was {0} / Now {1}"
+                                 .format(response['linux_profile'].admin_username, self.linux_profile.get('admin_username'))))
+                        to_be_updated = True
+                        # self.module.warn("linux_profile.admin_username cannot be updated")
+
+                    # Cannot have more that one agent pool profile for now
+                    if len(response['agent_pool_profiles']) != len(self.agent_pool_profiles):
+                       self.log("Agent Pool count is diff, need to updated")
+                       to_be_updated = True
+
+                    if response['kubernetes_version'] != self.kubernetes_version:
+                        to_be_updated = True
+
+                    for profile_result in response['agent_pool_profiles']:
+                        matched = False
+                        for profile_self in self.agent_pool_profiles:
+                            if profile_result['name'] == profile_self['name']:
+                                matched = True
+                                if profile_result['count'] != profile_self['count'] \
+                                    or profile_result['vm_size'] != profile_self['vm_size'] \
+                                    or profile_result['os_disk_size_gb'] != profile_self['os_disk_size_gb'] \
+                                    or profile_result['dns_prefix'] != profile_self['dns_prefix'] \
+                                    or profile_result['vnet_subnet_id'] != profile_self.get('vnet_subnet_id') \
+                                    or set(profile_result['ports'] or []) != set(profile_self.get('ports') or []):
+                                    self.log(("Agent Profile Diff - Origin {0} / Update {1}".format(str(profile_result), str(profile_self))))
+                                    to_be_updated = True
+                        if not matched:
+                            self.log("Agent Pool not found")
+                            to_be_updated = True
+
```

Comment:
Wondering can we do a & operator between response['agent_pool_profiles'] and self.agent_pool_profiles to decide to_be_updated true or false?

Reply:
there are some other field